data_structures/src/Stack/Stack.cs

```diff
@@ -26,36 +27,37 @@ public bool IsEmpty()
 
         public T Peek()
         {
-            if(!IsEmpty())            
-                return _items[_topOfStack -1];            
-            else            
-                throw new Exception("Stack is empty - Cannot peek.");            
+            if(IsEmpty())
+                throw new Exception("Stack is empty - Cannot peek.");
+
+            return _items[_top - 1];
         }
 
         public T Pop()
         {
-            if(!IsEmpty())
-            {
-                var itemToReturn = Peek();
-                _items[_topOfStack--] = default(T);
-                return itemToReturn;
-            }
-            else            
+            if(IsEmpty())
                 throw new Exception("Stack is null - cannot pop.");
-            
+
+            var itemToReturn = Peek();
+
+            _items[_top--] = default(T);
+            _itemCount--;
```

Comment:
We can get way without using the `_itemCount` variable, but readability suffers...